web/modules/custom/joinup_group/src/EventSubscriber/JoinupGroupOgSubscriber.php

```diff
@@ -41,4 +43,25 @@ public function provideOgGroupPermissions(OgPermissionEventInterface $event): vo
     );
   }
 
+  /**
+   * Gives moderators access to view, create, edit and delete all group content.
+   *
+   * @param \Drupal\og\Event\GroupContentEntityOperationAccessEventInterface $event
+   *   The event that fires when an entity operation is being performed on group
+   *   content.
+   */
+  public function moderatorsCanManageGroupContent(GroupContentEntityOperationAccessEventInterface $event): void {
+    $is_moderator = in_array('moderator', $event->getUser()->getRoles());
```

Comment:
I really don't like this! I'm repeating myself: don't check for roles, check for permissions. Isn't this equivalent to setting standard site-wide node permissions to `moderators`, such as `create news content`, `edit any document content`, etc? What I'm missing here? A site-wide permission should ***ALWAYS*** override an OG permission. It's possible that I don't get the purpose of this subscriber. If I'm right that the subscriber's job is to allow moderators to operate on group content and the site-wide permissions are not enough then It might be a bug in OG.

Reply:
This was already like this, this code has just moved from `joinup_og_user_access_alter()` to here, unchanged. This was necessary to accommodate the latest OG code which no longer fires this hook for entity operations.

Reply:
I looked at this a bit closer and it seems indeed to be unnecessary, this was probably needed to work around some oddities in the access calculation, but after the recent improvements in `OgAccess` this should no longer be required.

I have tentatively removed this entire section, let's see what the tests say.

Reply:
Code is gone, tests are still green! Goodbye :grin: